11 harmonize the nexus web pages convert the current wiki to use sphinx same as the main nexus documentation

RussBerg opened the pull request:
The old Vagrant based wiki files have been removed and replaced with a sphinx-build just like the standard nexus manual.
Closes #11 

Using *linkchecker* I managed to resolve the internal links such that linkchecker does not report any issues but when I look at **external** links then there are a great many that have issues such  as links to projects that used to exist when the markdown was edited, for example:  `Project on Github <https://github.com/nexusformat/NIAC/projects/3>`__ etc.

Being that the current wiki contains these broken links it is perhaps not an issue?, it would take significant effort to go through everything to remove/fix all of the link problems.

This will need some discussion at an upcoming NIAC technical meeting but before then this PR needs to be reviewed as it likely  needs a few iterations.

PeterC-DLS commented:
Some points:
- build directory included
- maybe leave the markdown and use a sphinx extension: https://www.sphinx-doc.org/en/master/usage/markdown.html - this will be a lot less churn
- consider deploying a specific website github pages-only repo then we can also add the historical versions of the manual in the same place (need to check what domains are used and whether we need to redirect old domains to sub-directories)

RussBerg commented:
> Some points:
> 
>     * build directory included
> 
>     * maybe leave the markdown and use a sphinx extension: https://www.sphinx-doc.org/en/master/usage/markdown.html - this will be a lot less churn
> 
>     * consider deploying a specific website github pages-only repo then we can also add the historical versions of the manual in the same place (need to check what domains are used and whether we need to redirect old domains to sub-directories)


1. build dir 👍 
2. I did that when I started working on this but the problem is that the the .md files essentially remain as is so the look is the old wiki not the goal of the PR which is to harmonize (unless I missed something), so running the .md through pandoc to convert them to .rst was the only way to maintain the look that matched the manual.
3. will do, I am just trying to get through the conversions and formatting first then will get your's and Pete's input as to where we want to put things.

When I get some time to finish off the formatting issues (thi sweek hopefully) I will convert this PR back from draft for you to have a look at.

PeterC-DLS commented:
I was thinking with my point 2 suggestion that using that sphinx extension allows you to leave the markdown content as-is (i.e. it does the conversion from `md` - as well as any `rst`) but render it in whatever sphinx theme we want.

RussBerg commented:
ok I will check again because this was months ago that I had tried it maybe I missed something

prjemian commented:
Here's another [repo](https://github.com/BCDA-APS/bluesky_training/blob/main/docs/source/conf.py) that builds its docs from .rst, .md, and .ipynb files.  It's `conf.py` file might be an example.  Also visit the repo's `pyproject.toml` for the dependencies.

RussBerg commented:
I must have missed something when I first looked at this its been so long I thought it was "the look" didn't change but the look is correct, regardless I am able to build the docs using **myst_parser** but there are numerous issues though, for instance the local file links do not render the href in such a way that the links work, instead it treats the local link as an **anchor** to the file it is referenced in. For example this is how myst renders the link specified in the .md: 

```  * [Facilities](Facilities.html "wikilink") using NeXus  ```

As:

```<li><p><a class="reference internal" href="#Facilities.html"><span class="xref myst">Facilities</span></a> using NeXus</p></li>```

I spent the better part of the morning trying to configure myst_parser to treat the links as local links but no joy, the only way i was able to solve it was to have a function connected to *app.connect('build-finished'...)*  walk all the generated html files and remove the '#' character from the href=. This approach works and does produce html files with links that work but during the sphinx build there are a ton of  **WARNING: 'myst' cross-reference target not found:** warnings (among others) that is just one example. 

I have been approaching this from a _build without warnings/errors_ but maybe I should ask **do we care about warnings during the build?** if not and we only care about functioning web pages then that would make this task much quicker, for me anyway.  

With the way I had been doing it (converting all .md to .rst) I am nearly done making all the files look nice (straight **pandoc** conversion from .md to .rst is not very readable), so I would prefer to just finish that off as opposed to trying to retain the .md files and sort out all of the issues which I don't think is going to save me any time, plus I had organized the files into some semblance of order as opposed to everything in the **content** directory which I also prefer. 

I freely admit that I'm no expert on this stuff so I don't have a problem if you guys want to take a crack at using the .md files I just think i am closer to being done the initial way and it more closely mirrors the manual documentation in terms of repo structure.

PeterC-DLS commented:
I had a quick look at that [link](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#markdown-link-syntax):
```
[Facilities](Facilities.html "wikilink") using NeXus
```
And it looks like `MyST` is not recognising that the first word of the link is an external target. There's two ways that may fix this:

1. Modify the link so it's recognised as an [external target](https://myst-parser.readthedocs.io/en/latest/syntax/cross-referencing.html#customising-external-url-resolution)
2. Use [linkify](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#linkify)

PeterC-DLS commented:
I've had a hack this afternoon and fixed a few issues:
1.  clone this repo at my #15 branch,
2. remove from `source` all of these directories -active_projects, codecamps,  content,  extra_files,  facilities,  faq,  img,  niac,  teleconferences from your branch
3. copy in the `content` from my branch
4. run two sed commands to correct some URLs:
```
# replace User URLs
sed -r -i~ -E -e '$!N;s/\[([^]]+)\]\(U.*html "wikilink"\)/\1 /;P;D' content/*.md
# replace .html of relative URLs
sed -r -i~ -E -e 's/\(([^/].+)\.html/\(\1.md/g' content/*.md
# repeat the last sed three more times
```
 5. paste at the end of `index.md` the contents of the attached file [toctree.txt](https://github.com/user-attachments/files/18606848/toctree.txt)
 6. build the web pages

It gives reasonable output but needs re-arranging into the hierarchy that you had to get the H1 headers in place

RussBerg commented:
to be clear I'm not saying that using the .md files and myst_parser wont work, 
I'm saying that I prefer:
- a structure that resembles the manual repo
- editing rst files 
- more organized file structure
of what I have in this PR

As I have mentioned there are still rst files that need a more readable format but I would like you and Pete to have a look at what I have and see if yawl agree with me or not :)

The one thing to note is that at the start of the build the logo files are retrieved from the nexusformat/NIAC repo so that they are always up to date with what the manual has.

PeterC-DLS commented:
Your new structure is great for organising things compared to the current single directory. However, there's one unstated use-case that I think most people add new items directly via GH's web interface so MD is rendered out-of-the-box there when adding new files.

PeterC-DLS commented:
On a side note, it seems the conversion of the site from a wiki to Jekyll built static site was incomplete and lots of wiki links such as `User:<blah>`, `Help:<blah>`, `Talk:<blah>`, `Media:<blah>`. See the output of
```
$ grep -R -E --include=*.md '\([^:]+:[^/]{2,}.*\.html' content/
```
when run on master.

PeterC-DLS commented:
Here's a quick review:

1. don't need the extra wiki directory so move `source` to top of repo
2. Title missing from landing page "About the NeXus Data Format"
3. Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
4. Index link at top leads to empty page
5. Left side bar is missing but nice to have its text somewhere
6. Update NIACDoc at some point
7. Translate trac tickets to GH issues (eg. in NIACActionItems, http://trac.nexusformat.org/definitions/ticket/11 -> https://github.com/nexusformat/definitions/issues/11)
8. Some missing content from examples when I compare output of `grep -r --include=*.md '<NXdetector' content/ | sort` with `grep -r --include=*.rst '<NXdetector' wiki/source | sort`. E.g. https://www.nexusformat.org/Connecting_NXdata_to_NXdetector.html with general_content/Connecting_NXdata_to_NXdetector.html

RussBerg commented:
> Here's a quick review:
> 
>     1. Don't need the extra wiki directory so move `source` to top of repo
> 
>     2. Title missing from landing page "About the NeXus Data Format"
> 
>     3. Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
> 
>     4. Index link at top leads to empty page
> 
>     5. Left side bar is missing but nice to have its text somewhere
> 
>     6. Update NIACDoc at some point
> 
>     7. Translate trac tickets to GH issues (eg. in NIACActionItems, http://trac.nexusformat.org/definitions/ticket/11 -> [Make a proposal for NXcharacterisation definitions#11](https://github.com/nexusformat/definitions/issues/11))
> 
>     8. Some missing content from examples when I compare output of `grep -r --include=*.md '<NXdetector' content/ | sort` with `grep -r --include=*.rst '<NXdetector' wiki/source | sort`. E.g. https://www.nexusformat.org/Connecting_NXdata_to_NXdetector.html with general_content/Connecting_NXdata_to_NXdetector.html

1. Don't need the extra wiki directory so move source to top of repo 
   :+1: 
2. Title missing from landing page "About the NeXus Data Format"
   :+1: 
3. Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
   **renamed** 
4. Index link at top leads to empty page
  **removed**
5. Left side bar is missing but nice to have its text somewhere
  **added** with logo
6. Update NIACDoc at some point
  **updated**
7. Translate trac tickets to GH issues (eg. in NIACActionItems, http://trac.nexusformat.org/definitions/ticket/11 ->
https://github.com/nexusformat/definitions/issues/11)
  Looks like this is served from ISIS? I have no access 
8. Some missing content from examples when I compare output of grep -r --include=*.md '<NXdetector' content/ | sort with grep -r --include=*.rst '<NXdetector' wiki/source | sort. E.g. https://www.nexusformat.org/Connecting_NXdata_to_NXdetector.html with general_content/Connecting_NXdata_to_NXdetector.html
  **added**

PeterC-DLS commented:
Some problems when I tested today:

- Doesn't build until I comment out https://github.com/nexusformat/wiki/blob/dabfb1ba7819cb44eda9d0f57c4b2bfcd6a5a86e/source/conf.py#L53 then I get this:
  > Warning, treated as error:
/home/peter/work/3rd/wiki.git/source/teleconferences/Telco_20250213.rst:39:Block quote ends without a blank line; unexpected unindent. [docutils]
- Several Telco minutes are missing, for example:
  - Telco_20250116.rst
  - Telco_20250127.rst
  - Telco_20250226.rst
- Formatting of at least one Telco pages seem off 
  - Telco_20240820.rst

RussBerg commented:
> Some problems when I tested today:
> 
>     * Doesn't build until I comment out https://github.com/nexusformat/wiki/blob/dabfb1ba7819cb44eda9d0f57c4b2bfcd6a5a86e/source/conf.py#L53
>        then I get this:
>       > Warning, treated as error:
>       > /home/peter/work/3rd/wiki.git/source/teleconferences/Telco_20250213.rst:39:Block quote ends without a blank line; unexpected unindent. [docutils]
> 
>     * Several Telco minutes are missing, for example:
>       
>       * Telco_20250116.rst
>       * Telco_20250127.rst
>       * Telco_20250226.rst
> 
>     * Formatting of at least one Telco pages seem off
>       
>       * Telco_20240820.rst

hi @PeterC-DLS 
- fixed warning on Telco_20250213.rst
- I migrated the minutes for 20250116 and 20250127 but I couldnt find the minutes for 20250226 which I think was when I asked Sandor to hold off updating minutes on the old wiki until this is merged then add them.
- Telco_20240820.rst was missing its content so retrieved from original .md file and added.

PeterC-DLS commented:
Other things I've noticed:

- missing single quotation marks aka apostrophes (these get replaced by three spaces - `grep "[a-z]   [a-z]"`)
- missing double quotation marks (these get replaces by four spaces `grep "[A-Za-z]    "`)

PeterC-DLS commented:
Ongoing fallout of md to rst conversion, I see formatting issues on the following pages:
- 2014_axes_and_uncertainites
- 2014_How_to_find_default_data
- Archive_Definitions has XML missing
- Associating_Axes_with_Data
- ConcordanceDiscussion: also extra entries in site_map page (ConcordanceSummary, The Variant Issue, etc)
- Containers_and_Inheritance
- Design_Issues: also incorrect links to Contents, NXmonochromator_and_inheritance and Units
- Detector_Geometry
- DLSraw: missing 1st sentence

Ok, on checking the first ten pages and finding issues - I believe there's problems on most of pages!

RussBerg commented:
@PeterC-DLS I agree with the discussion this morning on the concern over content that may be getting lost during translation, to that end I will go with the other option of a sphinx interpretation of the .md files instead, putting PR back into draft.

PeterC-DLS commented on source/.gitignore:
```diff
@@ -0,0 +1,5 @@
+_site
+.sass-cache
```
Most entries not relevant

RussBerg replied:
removed 👍

PeterC-DLS commented on source/LICENSE.md:
```diff
@@ -1,3 +1,6 @@
+LICENSE
```
Is this LICENSE relevant anymore?

RussBerg replied:
I'm not a licensing expert so I was going to leave that one up to you to decide :)

PeterC-DLS replied:
We don't use Hyde Jekyll anymore so this license does not apply

PeterC-DLS commented on source/content/2014_How_to_find_default_data.md:
```diff
@@ -3,6 +3,8 @@ title: 2014 How to find default data
 permalink: 2014_How_to_find_default_data.html
 layout: wiki
 ---
+2014 How to find default data
```
Is it worth remove the lines above this here and all other `md` files?

PeterC-DLS replied:
Ok, we can remove wiki type things in a later clean-up

PeterC-DLS reviewed (commented):
It's looking great. Some things I've noticed:
-  Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
- Index link at top leads to empty page. Perhaps link back contents section of landing page if you can't generate an index
- 'Telco_20250116 and Telco_20250127 are missing some content
- There's a number of links broken like this (as indicated in broken links file). Best to grep --include=*.md -r -E '\] \(' source/
- A number of missing images: mars.png, pinhole_2.png, slit.png

PeterC-DLS commented:
It's looking great. Some things I've noticed:
- Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
- Index link at top leads to empty page
- 'Telco_20250116 and Telco_20250127 are missing some content
- There's a number of links broken like this (as indicated in broken links file). Best to `grep --include=*.md -r -E '\] \(' source/`
- A number of missing images: mars.png, pinhole_2.png, slit.png

RussBerg commented on source/content/MonoXSingle.md:
```diff
@@ -3,6 +3,8 @@ title: MonoXSingle
 permalink: MonoXSingle.html
 layout: wiki
 ---
+MonoXSingle
```
basically on oll these "replicates subheading" issues that was due to a tool I wrote to go and add a title to each .md file to remove a warning/error from sphinx build so they were not done by hand

RussBerg commented on source/content/Teleconferences.md:
````diff
@@ -447,3 +454,145 @@ We aim to publish an agenda a few days in advance of the meetings.
 
 [Telco 20150114](Telco_20150114.html "wikilink") Wed, 14th Jan, 12:00 GMT
+
+```{toctree}
````
the build complains otherwise that the files are not included as part of a toctree

RussBerg commented on source/content/Teleconferences.md:
```diff
@@ -447,3 +454,145 @@ We aim to publish an agenda a few days in advance of the meetings.
+Telco_20250213.md
+Telco_20250226.md
+Telco_20250312.md
+Telco_20250324.md
```
"need" I dont think so but I've added them

RussBerg commented on source/index.md:
```diff
@@ -0,0 +1,171 @@
+content/Utilities.md
+content/Workshop_PSI_2010.md
+content/XESraw.md
+LICENSE.md
```
same as above

prjemian commented on source/content/Codecamps.md:
````diff
@@ -0,0 +1,20 @@
+
+```{toctree}
+:maxdepth: 2
+:caption: Contents:
````
Consider using [`:glob:`](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-option-toctree-glob) here and gather all current and future `CodeCamp*.md` entries together.  Assume same for `NIAC*.md`.

RussBerg replied:
thanks Pete, took your suggestion and tried it but was unable to make :glob: work regardless how I declared it in the markdown, it just would not match to the files resulting in "_WARNING: document isn't included in any toctree_" messages during build.

RussBerg commented:
> It's looking great. Some things I've noticed:
> 
> * Strange breadcrumb at top of page "nexusformat_wiki 1 documentation >> Documentation"
> * Index link at top leads to empty page
> * 'Telco_20250116 and Telco_20250127 are missing some content
> * There's a number of links broken like this [`[available for testing] (NeXus_42_Testing.html `](https://github.com/nexusformat/wiki/blob/e7abe879b6582dee47c950cd9d65a1460d55d07a/source/content/Latest_News.md?plain=1#L83) (as indicated in broken links file). Best to `grep --include=*.md -r -E '\] \(' source/`
> * A number of missing images: mars.png, pinhole_2.png, slit.png

1. Fixed, just needed to specify a blank html_title in conf.py
2. index has been removed with html_use_index = False
3. copied from main branch
4. those broken links pointed to files that did not exist in the wiki so I rendered them just as plain text so that they wouldnt a) show up in linkchecker output as well b) when looking that the web page they would not appear as valid links that lead to nowhere, c) I also thought it would provide an easy to understand indication when looking at the webpage that "_this should point to something_" and if someone had the knowledge of where the content for that webpage exists that the link originally pointed to they could add it, anyway that was the reasoning, I thought we had discussed those previously so I went that direction.
5. fixed case sensitive naming

PeterC-DLS commented:
Point 4 was mainly about broken links to things that exist (at least one for the example cited) - the aside to find all recently links you deliberately broke to pass the link checking. Many of those links could be removed in a later tidy-up.

PeterC-DLS commented:
Sorry, I misread the linked file as the release notes:
> Point 4 was mainly about broken links to things that exist (at least one for the example cited)

PeterC-DLS commented on source/content/NIAC.md:
```diff
@@ -149,8 +152,6 @@ Special Interest Groups: The NeXus International Advisory Committee will appoint
 Meetings
 --------
 
--   [NIAC2024](NIAC2024.html) NIAC Meeting, (hybrid)
```
Erroneous deletion these two lines?

PeterC-DLS commented on source/content/NIAC2006ILL.md:
```diff
@@ -24,16 +25,16 @@ as soon as possible.
-form](Help:Editing.html#Lists "wikilink"). If you want to add extended
-comments concerning any agenda item, please use this Wiki page's [ Talk
-Page](Talk:NIAC_Meetings.html "wikilink").
+form]((Editing#Lists.html "wikilink"). If you want to add extended
```
```suggestion
form] (Editing#Lists.html "wikilink"). If you want to add extended
```

PeterC-DLS commented on source/content/Teleconferences.md:
```diff
@@ -447,3 +454,146 @@ We aim to publish an agenda a few days in advance of the meetings.
+:maxdepth: 2
+:caption: Contents:
+
+Telco_20150114.md
```
Worth globbing these?

PeterC-DLS commented on source/index.md:
```diff
@@ -0,0 +1,168 @@
+  * [Facilities](content/Facilities.html "wikilink") using NeXus
+
+## Discussion and Development:
+* Next Meetings: [Autumn NIAC2024](content/NIAC2024.html "wikilink")
```
The current landing page has nothing listed for the next meetings; the item above should be moved to the list of recent meetings

PeterC-DLS commented on source/content/Associating_Axes_with_Data.md:
```diff
@@ -15,7 +17,7 @@ them as properties of the data. Now, this limitation of HDF has since
 covered by what we do today. Thus this page describes how to improve.
 Discussion comments should be placed on the [
-discussion](Talk:Associating_Axes_with_Data.html "wikilink") page.
+discussion](Associating_Axes_with_Data.html "wikilink") page.
```
This is self-referencing - simpler to replace with "this"

PeterC-DLS commented on source/content/NXdir.md:
```diff
@@ -43,7 +45,7 @@ compilation).
 Some common usages are:
 
--   Print the online help:
+-   Print the online (
```
```suggestion
-   Print the online (
```
```suggestion
-   Print the online
```

PeterC-DLS commented on source/content/NIAC2-22_fall_minutes.md:
```diff
@@ -3,6 +3,8 @@ title: Minutes of the NIAC2022 Fall Meeting
 permalink: NIAC2022_fall_minutes.html
 layout: wiki
 ---
+Minutes of the NIAC2022 Fall Meeting
```
Correct the name of this file

PeterC-DLS commented on source/_templates/google_search.html:
```diff
@@ -0,0 +1,35 @@
+{#
```
Is this actually used? I see in the html code but it's not visible

PeterC-DLS commented on source/content/NIAC.md:
```diff
@@ -149,8 +152,6 @@ Special Interest Groups: The NeXus International Advisory Committee will appoint
 Meetings
 --------
 
--   [NIAC2024](NIAC2024.html) NIAC Meeting, (hybrid)
```
Add this back in

RussBerg commented:
closes #2 
closes #3
